### PR TITLE
RUMM-139 Add documentation for logging

### PR DESCRIPTION
**Pull request by cswatt**
### What and why?

A short description of what changes this PR introduces and why.

### How?

A brief description of implementation details of this PR.

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [ ] Make sure each commit and the PR mention the Issue number or JIRA reference

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Datadog IOS SDK documentation
```
Comment (cswatt) [inline docs/README.md]:
```suggestion
# Datadog iOS SDK Documentation
```

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+Find in this folder dedicated documentation for:
+
+* [Collect and send logs from your IOS application to Datadog](/log_collection.md).
```
Comment (cswatt) [inline docs/README.md]:
```suggestion
* [Collecting and sending logs from your iOS application to Datadog](/log_collection.md).
```

File: docs/log_collection.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
+# IOS Log Collection
```
Comment (cswatt) [inline docs/log_collection.md]:
```suggestion
# iOS Log Collection
```

File: docs/log_collection.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+Send logs to Datadog from your IOS applications with [Datadog's `dd-sdk-ios` client-side logging library][1] and leverage the following features:
```
Comment (cswatt) [inline docs/log_collection.md]:
```suggestion
Send logs to Datadog from your iOS applications with [Datadog's `dd-sdk-ios` client-side logging library][1] and leverage the following features:
```

File: docs/log_collection.md [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+
+* Log to Datadog in JSON format natively.
+* Add `context` and extra custom attributes to each log sent.
+* Record real client IP addresses and User-Agents.
+* Optimized network usage with automatic bulk posts.
+
+**Note**: The `dd-sdk-ios` library supports all IOS versions from 11.
```
Comment (cswatt) [inline docs/log_collection.md]:
```suggestion
**Note**: The `dd-sdk-ios` library supports all iOS versions 11+.
```

File: docs/log_collection.md [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+    {{% tab "Swift Package Manager (SPM)" %}}
+
+To integrate using Apple's Swift package manager, add the following as a dependency to your Package.swift:
+```swift
+.package(url: "https://github.com/Datadog/dd-sdk-ios.git", .upToNextMajor(from: "0.0.0"))
+```
+
+    {{% /tab %}}
+    {{% tab "Carthage" %}}
+Not yet supported
```
Comment (cswatt) [inline docs/log_collection.md]: If there are no very immediate plans to support carthage, let's just remove this tab

File: docs/log_collection.md [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+
+    {{% /tab %}}
+    {{< /tabs >}}
+
+2. Initialize the library with your application context and your [Datadog client token][2]. For security reasons, you must use a client token: you cannot use [Datadog API keys][3] to configure the `dd-sdk-ios` library as they would be exposed client-side in the IOS application IPA byte code. For more information about setting up a client token, see the [client token documentation][2]:
```
Comment (cswatt) [inline docs/log_collection.md]:
```suggestion
2. Initialize the library with your application context and your [Datadog client token][2]. For security reasons, you must use a client token: you cannot use [Datadog API keys][3] to configure the `dd-sdk-ios` library as they would be exposed client-side in the iOS application IPA byte code. For more information about setting up a client token, see the [client token documentation][2]:
```

File: docs/log_collection.md [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+        .builderUsing(clientToken: config.clientToken)
+        .set(logsEndpoint: .eu)
+        .build()
+)
+```
+
+    {{% /tab %}}
+    {{< /tabs >}}
+
+When writing your application, you can enable development logs. All internal messages in the library with a priority equal or higher than the provided level are then logged to console logs.
```
Comment (cswatt) [inline docs/log_collection.md]:
```suggestion
When writing your application, you can enable development logs. All internal messages in the library with a priority equal to or higher than the provided level are then logged to console logs.
```

File: docs/log_collection.md [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+
+```swift
+    Datadog.verbosityLevel = .debug
+```
+
+3. Configure the IOS Logger:
```
Comment (cswatt) [inline docs/log_collection.md]:
```suggestion
3. Configure the iOS Logger:
```

File: docs/log_collection.md [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+
+## Advanced logging
+
+### Initialization
+
+The following methods in `Logger.Builder` can be used when initializing the logger to send logs to Datadog:
+
+| Method                           | Description                                                                                                                                                                                                                         |
+|----------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `sendNetworkInfo(_ enabled: true)`    | Add `network.client.connectivity` attribute to all log. The data logged by default is `connectivity` (`Wifi`, `3G`, `4G`...) and `carrier_name` (`AT&T - US`). |
```
Comment (cswatt) [inline docs/log_collection.md]:
```suggestion
| `sendNetworkInfo(_ enabled: true)`    | Add `network.client.connectivity` attribute to all logs. The data logged by default is `connectivity` (`Wifi`, `3G`, `4G`...) and `carrier_name` (`AT&T - US`). |
```

File: docs/log_collection.md [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+| `set(serviceName: <SERVICE_NAME>)` | Set `<SERVICE_NAME>` as value for the `service` [standard attribute][4] attached to all logs sent to Datadog.                                                                                                                        |
```
Comment (cswatt) [inline docs/log_collection.md]:
```suggestion
| `set(serviceName: <SERVICE_NAME>)` | Set `<SERVICE_NAME>` as the value for the `service` [standard attribute][4] attached to all logs sent to Datadog.                                                                                                                        |
```

File: docs/log_collection.md [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+[Learn more about Datadog tags][5].
+
+#### Global Attributes
+
+##### Add attributes
+
+By default, the following attributes are added to all logs sent by a logger:
+
+* `http.useragent` and its extracted `device` and `OS` properties
+* `network.client.ip` and its extracted geographical properties (`country`, `city`)
```
Comment (ncreated) [inline docs/log_collection.md]: Also:
```
"logger.version": "\(sdkVersion)", // Datadog SDK version
"logger.thread_name" : "main", // either "main" or "background"
"application.version": "1.0.0" // client's app version extracted from `Info.plist`
```

File: docs/log_collection.md [hunk 13]
```diff
@@ -0,0 +1,181 @@
...
+
+Use the `addAttribute(forKey key: "<ATTRIBUTE_KEY>", value: "<ATTRIBUTE_VALUE>")` function to add a custom attribute to all logs sent by a specific logger:
+
+```swift
+// This adds an attribute "device-model" with a string value
+logger.addAttribute(forKey: "device-model", value: UIDevice.current.model)
+```
+
+**Note**: `<ATTRIBUTE_VALUE>` can be any primitive, String, or Date.
```
Comment (ncreated) [inline docs/log_collection.md]:
```suggestion
**Note**: `<ATTRIBUTE_VALUE>` can be anything conforming to `Encodable` (String, Date, custom `Codable` data model, ...).
```

File: docs/log_collection.md [hunk 14]
```diff
@@ -0,0 +1,181 @@
...
+3. Configure the IOS Logger:
+
+    ```swift
+    logger = Logger.builder
+        .set(serviceName: "ios-sdk-example-app")
+        .printLogsToConsole(true, usingFormat: .shortWith(prefix: "[iOS App] "))
+        .build()
+    ```
+
+4. Send a custom log entry directly to Datadog with one of the following functions:
```
Comment (ncreated) [inline docs/log_collection.md]:
```suggestion
4. Send a custom log entry directly to Datadog with one of the following methods:
```

File: docs/log_collection.md [hunk 15]
```diff
@@ -0,0 +1,181 @@
...
+| `sendLogsToDatadog(_ enabled: true)`    | Set to `true` to send logs to Datadog.                                                                                                                                                                                              |
+| `set(loggerName: <LOGGER_NAME>)`   | Set `<LOGGER_NAME>` as the value for the `logger.name` attribute attached to all logs sent to Datadog.                                                                                                                                   |
+| `build()`                        | Build a new logger instance with all options set.                                                                                                                                                                                   |
+
+### Global configuration
+
+Find below functions to add/remove tags and attributes to all logs sent by a given logger.
```
Comment (ncreated) [inline docs/log_collection.md]:
```suggestion
Find below methods to add/remove tags and attributes to all logs sent by a given logger.
```

File: docs/log_collection.md [hunk 16]
```diff
@@ -0,0 +1,181 @@
...
+
+#### Global Tags
+
+##### Add Tags
+
+Use the `addTag(withKey: "<TAG_KEY>", value: "<TAG_VALUE>")` function to add tags to all logs sent by a specific logger:
```
Comment (ncreated) [inline docs/log_collection.md]:
```suggestion
Use the `addTag(withKey: "<TAG_KEY>", value: "<TAG_VALUE>")` method to add tags to all logs sent by a specific logger:
```

File: docs/log_collection.md [hunk 17]
```diff
@@ -0,0 +1,181 @@
...
+```swift
+// This adds a tag "build_configuration:debug"
+logger.addTag(withKey: "build_configuration", value: "debug")
+```
+
+**Note**: `<TAG_VALUE>` must be a String.
+
+##### Remove Tags
+
+Use the `removeTag(withKey key: "<TAG_KEY>")` function to remove tags from all logs sent by a specific logger:
```
Comment (ncreated) [inline docs/log_collection.md]:
```suggestion
Use the `removeTag(withKey key: "<TAG_KEY>")` method to remove tags from all logs sent by a specific logger:
```

File: docs/log_collection.md [hunk 18]
```diff
@@ -0,0 +1,181 @@
...
+#### Global Attributes
+
+##### Add attributes
+
+By default, the following attributes are added to all logs sent by a logger:
+
+* `http.useragent` and its extracted `device` and `OS` properties
+* `network.client.ip` and its extracted geographical properties (`country`, `city`)
+
+Use the `addAttribute(forKey key: "<ATTRIBUTE_KEY>", value: "<ATTRIBUTE_VALUE>")` function to add a custom attribute to all logs sent by a specific logger:
```
Comment (ncreated) [inline docs/log_collection.md]:
```suggestion
Use the `addAttribute(forKey key: "<ATTRIBUTE_KEY>", value: "<ATTRIBUTE_VALUE>")` method to add a custom attribute to all logs sent by a specific logger:
```

File: docs/log_collection.md [hunk 19]
```diff
@@ -0,0 +1,181 @@
...
+```swift
+// This adds an attribute "device-model" with a string value
+logger.addAttribute(forKey: "device-model", value: UIDevice.current.model)
+```
+
+**Note**: `<ATTRIBUTE_VALUE>` can be any primitive, String, or Date.
+
+##### Remove attributes
+
+Use the `removeAttribute(forKey key: "<ATTRIBUTE_KEY>")` function to remove a custom attribute from all logs sent by a specific logger:
```
Comment (ncreated) [inline docs/log_collection.md]:
```suggestion
Use the `removeAttribute(forKey key: "<ATTRIBUTE_KEY>")` method to remove a custom attribute from all logs sent by a specific logger:
```

File: docs/log_collection.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (hugokash) [reply docs/log_collection.md]: done

File: docs/log_collection.md (hunk 12 above)
Comment (hugokash) [reply docs/log_collection.md]: Done

File: docs/log_collection.md [hunk 20]
```diff
@@ -0,0 +1,189 @@
+# iOS Log Collection
+
+Send logs to Datadog from your iOS applications with [Datadog's `dd-sdk-ios` client-side logging library][1] and leverage the following features:
+
+* Log to Datadog in JSON format natively.
+* Add `context` and extra custom attributes to each log sent.
+* Record real client IP addresses and User-Agents.
+* Optimized network usage with automatic bulk posts.
+
+**Note**: The `dd-sdk-ios` library supports all iOS versions 12+.
```
Comment (iGranDav) [inline docs/log_collection.md]: 11+ now right?

Comment (ncreated) [reply docs/log_collection.md]: Ha! 🏅

Comment (ncreated) [inline docs/log_collection.md]:
```suggestion
**Note**: The `dd-sdk-ios` library supports all iOS versions 11+.
```

File: docs/log_collection.md [hunk 21]
```diff
@@ -0,0 +1,189 @@
...
+## Setup
+
+1. Declare the library as a dependency depending on your package manager:
+
+    {{< tabs >}}
+    {{% tab "CocoaPods" %}}
+
+You can use [CocoaPods][6] to install `dd-sdk-ios`:
+```
+pod 'Datadog'
```
Comment (ncreated) [inline docs/log_collection.md]:
```suggestion
pod 'DatadogSDK'
```

**Pull request merged**